dask/dataframe/io/parquet/core.py

```diff
@@ -684,4 +605,172 @@ def apply_filters(parts, statistics, filters):
     return parts, statistics
 
 
+def process_statistics(parts, statistics, filters, index, chunksize):
+    """Process row-group column statistics in metadata
+    Used in read_parquet.
+    """
+    index_in_columns = False
+    if statistics:
+        result = list(
+            zip(
+                *[
+                    (part, stats)
+                    for part, stats in zip(parts, statistics)
+                    if stats["num-rows"] > 0
+                ]
+            )
+        )
+        parts, statistics = result or [[], []]
+        if filters:
+            parts, statistics = apply_filters(parts, statistics, filters)
+
+        # Aggregate parts/statistics if we are splitting by row-group
+        if chunksize:
+            parts, statistics = aggregate_row_groups(parts, statistics, chunksize)
```

Comment:
Do we need statistics after this line?  My guess is that we can figure out the sorted columns before we aggregate (aggregating will make this no worse), and that way we don't need to merge the statistics, which might simplify things considerably.

Reply:
The aggregation was being done before the call to `sorted_columns`, because that function will effectively calculate the divisions.  But you are correct, we can also let the divisions be calculated incorrectly and them adjust them during aggregation (might be more efficient than merging statistics)

Reply:
Ah, if it's necessary then let's keep it.  Mostly I was looking for ways to simplify things.  I suspect that for performance everything here is effectively free.